### DISCUSSION TITLE
Why do I get unexpected results when trying to control the height of a NumericUpDown control?

### DISCUSSION THREAD

**Discussion by IronEaterBeast** [AvaloniaUI/Avalonia#15296] [Q&A]
The default size of the control is a bit too tall, so I tried using Height='25' to control its height. However, it didn't get smaller but was clipped instead. Is there any other way to control the height?

before:
![2024-04-10 103250](https://github.com/AvaloniaUI/Avalonia/assets/9688223/574faf92-92a3-4510-a135-115738961f77)

after:
![2024-04-10 103332](https://github.com/AvaloniaUI/Avalonia/assets/9688223/ced732f9-cfac-42d5-9d67-d43a40177bd1)

**Comment by stevemonaco** [comment]
It's being clipped because the control can't be resized that small. The layout container uses the size you provided, but the control itself needs more room to render. This overflows the available space and gets clipped.

Many controls contain fixed size elements, such as each up/down arrow in [ButtonSpinner](https://github.com/AvaloniaUI/Avalonia/blob/c466e960d1947d0adb745532faacbe9f97507f63/src/Avalonia.Themes.Fluent/Controls/ButtonSpinner.xaml#L99) (a part of `NumericUpDown`, Fluent theme) is exactly 16x8 pixels. You often need to both read the original theme's style for the control(s) and use DevTools to figure out what parts are fixed (or have minimum sizes) and what parts can be adjusted via Margin, Padding, or resource overrides. In the worst case, you may need to entirely replace the `ControlTemplate`.

**Reply by IronEaterBeast** [reply]
Even though I hide the arrows, the remaining text box still appears significantly taller than the text inside it.

**Reply by timunie** [reply]
use DevTools F12 to check the visual tree of that control. Also worth to lookup original styles

**Comment by rabbitism** [comment]
Yes this is Semi.Avalonia style issue. Thanks for reporting.

**Reply by rabbitism** [reply]
https://github.com/irihitech/Semi.Avalonia/discussions/339
Please check above discussion if you encountered this in Semi.Avalonia theme.
